src/executables/master.c

```diff
@@ -429,6 +432,27 @@ css_accept_old_request (CSS_CONN_ENTRY * conn, unsigned short rid, SOCKET_QUEUE_
     }
 }
 
+static int
+receive_server_info (CSS_CONN_ENTRY * conn, unsigned short rid, std::string & dbname, SERVER_TYPE & type)
+{
+  int name_length;
+  char *buffer = NULL;
+
+  if (css_receive_data (conn, rid, &buffer, &name_length, -1) == NO_ERRORS)
+    {
+      // *INDENT-OFF*
+      type = static_cast<SERVER_TYPE> ((int) buffer[0] - '0');
+      // *INDENT-ON*
+      dbname = buffer + 1;
```

Comment:
I think the length of the string has to be `name_length - 1`
```suggestion
      dbname = std::string (buffer + 1, name_length - 1);
```